src/connector.js

```diff
@@ -236,47 +240,51 @@ class BotiumConnectorSimpleSocketIO {
     const startPromises = []
     if (this.caps[Capabilities.SIMPLESOCKETIO_COOKIE_AUTOFILL] && (!this.socketOptions.transports || this.socketOptions.transports.includes('polling'))) {
       startPromises.push(
-        new Promise((resolve) => {
-          let resolved = false
-          let counter = 0
-          this.socket.io.on('open', () => {
-            this._debug('Received \'open\' event')
-            this.socket.io.engine.transport.on('pollComplete', () => {
-              this._debug('Received \'pollComplete\' event')
-              if (resolved) {
-                this._debug(`Cookie autofill is already resolved. 'socket.io.opts.extraHeaders': ${JSON.stringify(this.socket.io.opts.extraHeaders, null, 2)}`)
-                return
-              }
-              counter++
-              if (counter >= 3) {
-                this._debug('Failed to find cookies during \'pollComplete\' event')
+        new Promise((resolve, reject) => {
+          try {
+            let resolved = false
+            let counter = 0
+            this.socket.io.on('open', () => {
+              this._debug('Received \'open\' event')
+              this.socket.io.engine.transport.on('pollComplete', () => {
+                this._debug('Received \'pollComplete\' event')
+                if (resolved) {
+                  this._debug(`Cookie autofill is already resolved. 'socket.io.opts.extraHeaders': ${JSON.stringify(this.socket.io.opts.extraHeaders, null, 2)}`)
```

Comment:
maybe not just ignore it, but setting the new cookies to the socket header ? for future polling or HTTP requests ?